metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -787,24 +789,34 @@ def _find_dataflow_recipe(
         time_range_constraint: Optional[TimeRangeConstraint] = None,
     ) -> Optional[DataflowRecipe]:
         linkable_specs = linkable_spec_set.as_tuple
+        potential_source_nodes: Sequence[BaseOutput]
+        use_time_spine_source_node = False
         if measure_spec_properties:
             source_nodes = self._source_nodes
-            potential_source_nodes: Sequence[BaseOutput] = self._select_source_nodes_with_measures(
+            potential_source_nodes = self._select_source_nodes_with_measures(
                 measure_specs=set(measure_spec_properties.measure_specs), source_nodes=source_nodes
             )
             default_join_type = SqlJoinType.LEFT_OUTER
         else:
             # Only read nodes can be source nodes for queries without measures
-            source_nodes = self._read_nodes
-            source_nodes_to_linkable_specs = self._select_read_nodes_with_linkable_specs(
-                linkable_specs=linkable_spec_set, read_nodes=source_nodes
-            )
-            potential_source_nodes = list(source_nodes_to_linkable_specs.keys())
+            source_nodes = list(self._read_nodes)
+            potential_source_nodes = self._select_read_nodes_with_linkable_specs(
+                linkable_specs=linkable_spec_set, read_nodes=self._read_nodes
+            )
+            # `metric_time` does not exist if there is no metric in the query.
+            # In that case, we'll use the time spine table to represent `metric_time` values.
+            requested_metric_time_specs = [
+                time_dimension_spec
+                for time_dimension_spec in linkable_spec_set.time_dimension_specs
+                if time_dimension_spec.element_name == self._metric_time_dimension_reference.element_name
+            ]
+            if requested_metric_time_specs:
+                # Add time_spine source node to potential source nodes
+                potential_source_nodes = list(potential_source_nodes) + [self._time_spine_source_node]
+                use_time_spine_source_node = True
```

Comment:
This "tag a boolean local and then use it down the line" thing feels really brittle. I suspect we need a little refactor in this method to account for it properly, or maybe we should push this selection down into the node processor.

A smaller improvement would be to have the boolean just be an optional pointer to the time spine source node and then use it if set, something like:

```
input_time_spine_source_node: Optional[ReadSqlSourceNode] = None  # or whatever type it is

....

    if requested_metric_time_specs:
        ...
        input_time_spine_source_node = self._time_spine_source_node
```

Reply:
Will do!